File: Programmers/덧칠하기.py
def solution(n, m, section):
    answer = 0
    wall = [1 for _ in range(n)]
    rest = []
  
    for i in section:
        wall[i-1] = 0
      
    for i in range(len(wall)-m+1):
        if wall[i] == 0:
            for j in range(m):
                wall[i+j] = 1
            answer += 1
          
    for i in range(len(wall)-m+1, len(wall)):
        rest.append(wall[i])
      
    if 0 in rest:
        answer += 1
      
    return answer

File: Programmers/덧칠하기.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int solution(int n, int m, int section[], int section_count) {
    int answer = 0;

    int *wall = (int*)malloc(n * sizeof(int));
    memset(wall, 1, n * sizeof(int));

    for (int i = 0; i < section_count; i++) {
        wall[section[i] - 1] = 0;
    }

    for (int i = 0; i <= n - m; i++) {
        if (wall[i] == 0) {
            for (int j = 0; j < m; j++) {
                wall[i + j] = 1;
            }
            answer++;
        }
    }

    for (int i = n - m + 1; i < n; i++) {
        if (wall[i] == 0) {
            answer++;
            break;
        }
    }

    free(wall);
  
    return answer;
}
